### PR TITLE
Move zng open code from search into space

### PR DESCRIPTION
**Pull request by aswan**
Based on suggestions from @alfred-landrum while reviewing the
time indexing branch, peeled off from there.

### CODE REVIEW

File: zqd/search/search.go [hunk 1]
```diff
@@ -49,27 +45,18 @@ func NewSearch(ctx context.Context, s *space.Space, req api.SearchRequest) (*Sea
...
-		if !os.IsNotExist(err) {
-			return nil, err
-		}
-		f = ioutil.NopCloser(strings.NewReader(""))
-	}
-	zngReader, err := detector.LookupReader("bzng", f, resolver.NewContext())
+
+	zngReader, err := s.OpenZng(query.Span)
 	if err != nil {
-		f.Close()
```
Comment (henridf) [inline zqd/search/search.go:62]: We still need to close zngReader here

File: zqd/search/search.go [hunk 2]
```diff
@@ -49,27 +45,18 @@ func NewSearch(ctx context.Context, s *space.Space, req api.SearchRequest) (*Sea
...
 		return nil, err
 	}
 	zctx := resolver.NewContext()
 	mapper := scanner.NewMapper(zngReader, zctx)
 	mux, err := launch(ctx, query, mapper, zctx)
 	if err != nil {
-		f.Close()
```
Comment (henridf) [inline zqd/search/search.go:69]: we still need to close zngReader here

Comment (aswan) [reply zqd/search/search.go:69]: whoops, rebase slopiness, thanks for catching that.

File: zqd/search/search.go [hunk 3]
```diff
@@ -48,12 +48,14 @@ func NewSearch(ctx context.Context, s *space.Space, req api.SearchRequest) (*Sea
 
 	zngReader, err := s.OpenZng(query.Span)
 	if err != nil {
+		f.Close()
```
Comment (henridf) [inline zqd/search/search.go:51]: I don't think this one is needed

**Pull request merged**